sympy/printing/printer.py

```diff
@@ -55,135 +61,111 @@
         |
         `-- p._print_Basic(expr)
 
-    if ._print_Rational method exists in the printer, then it is called,
-    and the result is returned back.
-
-    otherwise, we proceed with trying Rational bases in the inheritance
-    order.
-
-3. As fall-back use the emptyPrinter method for the printer.
+    if ``._print_Rational`` method exists in the printer, then it is called,
+    and the result is returned back. Otherwise, the printer tries to call
+    ``._print_Number`` and so on.
 
-    As fall-back self.emptyPrinter will be called with the expression. If
-    not defined in the Printer subclass this will be the same as str(expr).
-"""
+3. **As a fall-back use the emptyPrinter method for the printer.**
 
-from __future__ import print_function, division
+    As fall-back ``self.emptyPrinter`` will be called with the expression. If
+    not defined in the Printer subclass this will be the same as ``str(expr)``.
 
-from sympy import Basic, Add
-
-from sympy.core.core import BasicMeta
+Example of Custom Printer
+^^^^^^^^^^^^^^^^^^^^^^^^^
 
-from functools import cmp_to_key
+.. _printer_example:
 
+In the example below, we have a printer which prints derivative of a function
```

Comment:
-> 
In the example below, we have a printer which prints the derivative of a function
in a shorter form.

(added "the" and dropped redundant reference to "code below".)